### PR TITLE
Components fix

### PR DESCRIPTION
**Pull request by idiotsiboon**
plese check the code

### CODE REVIEW

**Comment by idiotsiboon** [comment]
@vramana

File: js/components/loginScreen.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  renderPasswordFeild: function () {
+
+  },
+
+  render() {
+    return (
+      <h1>
+        <a className="btn btn-link" onClick={this.handleLogin}>Login</a>
+      </h1>
+          );
```
Comment (vramana) [inline js/components/loginScreen.js]: You should align the closing brace with the `r` of `return`

File: js/components/nav.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <a onClick={this.handleLogin} className="btn btn-link">Login</a>
+    );
+  },
+
+  renderLinks: function () {
+    return (
+      <span>
+        <a href="#" className="btn btn-link">Home</a>
+      </span>
+  );
```
Comment (vramana) [inline js/components/nav.js]: Same here.

File: js/components/nav.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+      <header className="navbar" id="NavId">
+        {this.renderBrand()}
+        <section className="navbar-section">
+          {this.renderLinks()}
+          {this.renderAvatar()}
+          {this.renderSignout()}
+        </section>
+      </header>
+  );
```
Comment (vramana) [inline js/components/nav.js]: Same here

File: js/components/homeScreen.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+}];
+
+/*
+var Nav = require("./nav.js");
+var Forum = require("./forum.js");
+*/
+window.HomeScreen = React.createClass({
+//  nav
+  renderNav: function () {
+    return (<Nav setRoute={this.props.setRoute} />);
```
Comment (vramana) [inline js/components/homeScreen.js]: Paranthesis here are unnecessary. `return <Nav setRoute={this.props.setRoute} />;` is enough.

File: js/components/homeScreen.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  },
+
+//  forum
+  renderForum: function () {
+    return (
+      <div className="grid-center">
+        <div className="col-8" >
+          <Forum posts={PostsJson} />
+        </div>
+      </div>);
```
Comment (vramana) [inline js/components/homeScreen.js]: Paranthesis in the next line.

File: js/components/homeScreen.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  },
+
+  render() {
+    return (
+      <div>
+  {this.renderNav()}
+  {this.renderForum()}
```
Comment (vramana) [inline js/components/homeScreen.js]: Not aligned correctly.

File: js/components/forum.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <div className="form-group">
+          <textarea
+            className="form-input"
+            value={this.state.content}
+            onChange={this.handleChange}
+            rows="3"
+          >
+          </textarea>
```
Comment (vramana) [inline js/components/forum.js:119]: `<Component prop1={prop1}></Component>` is same as `<Component prop1={prop1} />`.

So this is enough.

``` js
          <textarea
            className="form-input"
            value={this.state.content}
            onChange={this.handleChange}
            rows="3"
          />
```

File: js/components/app.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+/*
+var LoginScreen = require("./loginScreen.js");
```
Comment (vramana) [inline js/components/app.js]: We actually need a bundler like webpack. But I want to keep things simple. So. let's avoid that for a litte while more.

File: js/components/forum.js (hunk 7 above)
Comment (idiotsiboon) [inline js/components/forum.js:119]: the error is : expected JSX closing tag..is it fine?

Comment (vramana) [inline js/components/forum.js:119]: Wierd. Okay leave it like before then.

File: js/components/app.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+
+  render() {
+    const route = this.state.route;
+
+    switch (route) {
+
+      case "login":
+        return (<LoginScreen setRoute={this.setRoute} />);
+        break;
```
Comment (vramana) [inline js/components/app.js]: No need for break after return.

File: js/components/app.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+      case "homeScreen":
+        return (<HomeScreen setRoute={this.setRoute} />);
+        break;
```
Comment (vramana) [inline js/components/app.js]: Same here.

File: js/components/app.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (vramana) [inline js/components/app.js]: try adding this line here. Should fix a few errors.

 `/* global LoginScreen, HomeScreen, NotFoundScreen */`

File: js/components/forum.js (hunk 7 above)
Comment (idiotsiboon) [inline js/components/forum.js:119]: html is not rendering ..its true for a component..but this is a HTML tag here..I think.

Comment (vramana) [inline js/components/forum.js:119]: No. I write `<div {...props} />` all the time. I will figure it out once the PR is in.

File: js/components/forum.js [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+      </span>
+    );
+  }
+});
+
+/* START PostS*/
+var PostsList = React.createClass({
+  render: function () {
+    var x = this.props.posts.sort(
+        function (a,b) {
```
Comment (vramana) [inline js/components/forum.js]:
``` diff
- (a,b)
+ (a, b)
```

File: js/components/homeScreen.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
+/* global Nav,Forum */
```
Comment (vramana) [inline js/components/homeScreen.js:1]: Add `/* eslint-disable */` here.

File: js/components/homeScreen.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+  content: "Morbi non lectus. Aliquam sit amet diam in magna bibendum imperdiet. Nullam orci pede, venenatis non, sodales sed, tincidunt eu, felis.",
+  username: "Lawrence Howard",
+  createdAt: "7/27/2015"
+}, {
+  _id: "1cbd224e-1bfc-4de9-8b82-9124281230c3",
+  content: "Morbi porttitor lorem id ligula. Suspendisse ornare consequat lectus. In est risus, auctor sed, tristique in, tempus sit amet, sem.\n\nFusce consequat. Nulla nisl. Nunc nisl.",
+  username: "Judy Morales",
+  createdAt: "8/3/2015"
+}];
+
```
Comment (vramana) [inline js/components/homeScreen.js]: Add `/* eslint-enable */` here.

File: js/components/registerScreen.js [hunk 15]
```diff
@@ -0,0 +1,29 @@
+/*
+sername
+password
+abot
+handleregister
+*/
+
+var Register = React.createClass({
```
Comment (vramana) [inline js/components/registerScreen.js]: Make this `window.Register` and we are good to merge this PR.

**Comment by vramana** [comment]
:tada: All tests green.

**Pull request merged**